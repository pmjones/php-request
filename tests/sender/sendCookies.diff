001+ Fatal error: Uncaught TypeError: Argument 3 passed to SapiResponse::setCookie() must be of the type int, array given, called in /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCookies.php on line 7 and defined in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php:100
002+ Stack trace:
003+ #0 /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCookies.php(7): SapiResponse->setCookie('cookie4', 'value4', Array)
004+ #1 {main}
005+   thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php on line 100
001- array(5) {
002-   [0]=>
003-   string(30) "Set-Cookie: cookie1=v1%26%25v2"
004-   [1]=>
005-   string(26) "Set-Cookie: cookie2=v3&%v4"
006-   [2]=>
007-   string(122) "Set-Cookie: cookie3=value3; expires=Fri, 13-Feb-2009 23:31:30 GMT; Max-Age=0; path=/path; domain=doma.in; secure; HttpOnly"
008-   [3]=>
009-   string(136) "Set-Cookie: cookie4=value4; expires=Fri, 13-Feb-2009 23:31:30 GMT; Max-Age=0; path=/path; domain=doma.in; secure; HttpOnly; SameSite=lax"
010-   [4]=>
011-   string(75) "Set-Cookie: empty=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0"
012- }