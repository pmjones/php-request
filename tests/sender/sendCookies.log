
---- EXPECTED OUTPUT
array(5) {
  [0]=>
  string(30) "Set-Cookie: cookie1=v1%26%25v2"
  [1]=>
  string(26) "Set-Cookie: cookie2=v3&%v4"
  [2]=>
  string(122) "Set-Cookie: cookie3=value3; expires=Fri, 13-Feb-2009 23:31:30 GMT; Max-Age=0; path=/path; domain=doma.in; secure; HttpOnly"
  [3]=>
  string(136) "Set-Cookie: cookie4=value4; expires=Fri, 13-Feb-2009 23:31:30 GMT; Max-Age=0; path=/path; domain=doma.in; secure; HttpOnly; SameSite=lax"
  [4]=>
  string(75) "Set-Cookie: empty=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0"
}
---- ACTUAL OUTPUT
Fatal error: Uncaught Error: Call to undefined method SapiResponse::getHeaderCallback() in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php:20
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php(11): SapiResponseSender->registerHeaderCallback(Object(SapiResponse))
#1 /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCookies.php(15): SapiResponseSender->send(Object(SapiResponse))
#2 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponseSender.php on line 20
---- FAILED
