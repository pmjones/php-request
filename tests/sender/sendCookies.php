<?php
$response = new SapiResponse();
$response->setCookie('cookie1', 'v1&%v2');
$response->setRawCookie('cookie2', 'v3&%v4');
$response->setCookie('cookie3', 'value3', 1234567890, "/path", "doma.in", true, true);
$response->setCookie('cookie4', 'value4', [
    'expires' => 1234567890,
    'path' => "/path",
    'domain' => "doma.in",
    'secure' => true,
    'httponly' => true,
    'samesite' => 'lax',
]);
$response->setCookie('empty');
(new SapiResponseSender())->send($response);
var_dump(headers_list());
