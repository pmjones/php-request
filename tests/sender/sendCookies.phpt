--TEST--
SapiResponseSender::sendCookies
--CGI--
--INI--
expose_php=0
--FILE--
<?php
$response = new SapiResponse();
$response->setCookie('cookie1', 'v1&%v2');
$response->setRawCookie('cookie2', 'v3&%v4');
$response->setCookie('cookie3', 'value3', 1234567890, "/path", "doma.in", true, true);
$response->setCookie('cookie4', 'value4', [
    'expires' => 1234567890,
    'path' => "/path",
    'domain' => "doma.in",
    'secure' => true,
    'httponly' => true,
    'samesite' => 'lax',
]);
$response->setCookie('empty');
(new SapiResponseSender())->send($response);
--EXPECTHEADERS--
Set-Cookie: cookie1=v1%26%25v2
Set-Cookie: cookie2=v3&%v4
Set-Cookie: cookie3=value3; expires=Fri, 13-Feb-2009 23:31:30 GMT; Max-Age=0; path=/path; domain=doma.in; secure; HttpOnly
Set-Cookie: cookie4=value4; expires=Fri, 13-Feb-2009 23:31:30 GMT; Max-Age=0; path=/path; domain=doma.in; secure; HttpOnly; SameSite=lax
Set-Cookie: empty=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; Max-Age=0
--EXPECT--
