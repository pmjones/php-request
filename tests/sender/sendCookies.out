Fatal error: Uncaught TypeError: Argument 3 passed to SapiResponse::setCookie() must be of the type int, array given, called in /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCookies.php on line 7 and defined in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php:100
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/tests/sender/sendCookies.php(7): SapiResponse->setCookie('cookie4', 'value4', Array)
#1 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php on line 100